inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -491,10 +491,6 @@ bool is_exec_graph(const ngraph::Function& f) {
 
 bool resolve_dynamic_shapes(const ngraph::Function& f) {
     const auto & f_ops = f.get_ordered_ops();
-    if (std::all_of(f_ops.begin(), f_ops.end(),
-            [](std::shared_ptr<Node> results) { return !results->is_dynamic(); })) {
```

Comment:
I refreshed my knowledge with an email you've attached. And yes, you are right that in some cases nodes can always produce dynamic outputs even with static inputs. But removing this if statement we will apply `resolve_dynamic_shapes` for all functions including static. I suggest to check here both inputs and outputs node ports and only if some of them is dynamic then we go next.

Reply:
Yes, this is better solution. I'll talk tomorrow with @bszmelcz and explain.